src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -102,23 +115,21 @@ def tryConnect(self, host, port):
                 s.settimeout(None)
                 s.close()
 
-            i += 1
+            if connected:
+                break
 
         return connected
 
     @private
     def getStarted(self, service):
         max_tries = 3
+        started = False
 
-        started = self.middleware.call_sync('service.started', self.name)
-        if started is True:
-            return started
-
-        i = 0
-        while i < max_tries:
-            time.sleep(1)
+        for i in range(0, max_tries):
             started = self.middleware.call_sync('service.started', self.name)
-            i += 1
+            if started:
+                break
+            time.sleep(1)
```

Comment:
Why is `service` passed as a param but never used? should it call self.middleware.call_sync('service.started', service)?

Also:

```
if self.middleware.call_sync('service.started', self.name):
    return True
```

Would be simpler, with a return False at the end.

Reply:
Ok